lightning/src/ln/functional_test_utils.rs

```diff
@@ -89,6 +91,27 @@ impl<'a, 'b> Drop for Node<'a, 'b> {
 			assert!(self.node.get_and_clear_pending_msg_events().is_empty());
 			assert!(self.node.get_and_clear_pending_events().is_empty());
 			assert!(self.chan_monitor.added_monitors.lock().unwrap().is_empty());
+
+			// Check that if we serialize and then deserialize all our channel monitors we the same
+			// set of outputs to watch for on chain as we have now. Note that if we write tests
+			// that fully close channels and remove the monitors at some point this may break.
+			let new_watch = Arc::new(chaininterface::ChainWatchInterfaceUtil::new(Network::Testnet, Arc::clone(&self.logger) as Arc<Logger>));
+			let feeest = Arc::new(test_utils::TestFeeEstimator { sat_per_kw: 253 });
+			let new_monitor = test_utils::TestChannelMonitor::new(new_watch.clone(), self.tx_broadcaster.clone(), self.logger.clone(), feeest);
+			let old_monitors = self.chan_monitor.simple_monitor.monitors.lock().unwrap();
+			for (_, monitor) in old_monitors.iter() {
+				let mut w = test_utils::TestVecWriter(Vec::new());
+				monitor.write_for_disk(&mut w).unwrap();
+				let (_, new_mon) = <(Sha256d, ChannelMonitor<EnforcingChannelKeys>)>::read(
```

Comment:
(It's unfortunate there's so much overloading of "monitor". It makes it a bit tough to read this test.)

To better reflect what is tested against ("new" and "old" aren't accurate terms, IMHO), I'd consider the following local renaming:

s/new_watch/chain_monitor
s/new_monitor/channel_monitor
s/new_mon/deserialized_monitor

And then inlining the expression for `old_monitor` where it is used.